src/test/java/org/cactoos/io/ZipTest.java

```diff
@@ -75,14 +76,21 @@ public void mustZipEmptyDirectory() throws Exception {
                         new File(folder.getParentFile(), "empty.zip")
                     )
                     ) {
-                        return zipped.stream().map(
-                            ZipEntry::toString
-                        ).findFirst().get();
+                        return new IterableOf<>(
+                            new ListOf<>(
+                                new Mapped<>(
```

Comment:
@fanifieiev please use directly the iterable abstraction, you don't need list:
```
return new Mapped<>(
  ZipEntry::getName,
  zipped.stream()::iterator
)
```

should work

Reply:
@victornoel I need ListOf, because I need to collect all the values into collection before I leave the try block, otherwise it fails to get the values.
I would not do like that, if I could use just as you posted.

Reply:
@fanifieiev in that case please remove the wrapping `IterableOf`, it is not needed (a `List` is an `Iterable`).

Reply:
@victornoel Right, but `IterableOf `has equals that can check the equality of each element, the `ListOf `does not have.

Reply:
@fanifieiev have you tried? `ListOf` extends `ListEnvelope` which delegate the `equals` method so it should work as expected IMHO or we should change the matcher so that it works.

Reply:
@victornoel Have you see the implementation of `equals` of `ListOf ` ???
The implementation of `equals `for `ListOf` is here https://github.com/yegor256/cactoos/blob/master/src/main/java/org/cactoos/iterable/IterableEnvelope.java

Reply:
@fanifieiev yes, it delegates to the wrapped list, that's what I said

Reply:
@victornoel The warpped list in this case is `LinkedList`. It does not check the equality of each element.

Reply:
@victornoel Ok, will try to change the matcher

Reply:
> The warpped list in this case is LinkedList. It does not check the equality of each element.

@fanifieiev yes it does… that's part of the contract of equals on all `List`… if you don't want to test for order, use a `SetOf` instead

Reply:
@victornoel You are right, AbstractList has equals implementation. I though it did not, because it did not work for me.